### PR TITLE
Add media lib option to WYSIWYG

### PR DESCRIPTION
**Pull request by soupette**
#### Description of what you did:
This PR allows the use of the medial library in the wysiwyg component
To be merged after #5763

### CODE REVIEW

File: docs/3.0.0-beta.x/guides/registering-a-field-in-admin.md [hunk 1]
```diff
@@ -40,21 +40,99 @@ Once this step is over all we need to do is to create our new WYSIWYG which will
...
+  const [data, setData] = useState(null);
+  const [isDisplayed, setIsDisplayed] = useState(false);
+
+  useEffect(() => {
+    if (isOpen) {
+      setIsDisplayed(true);
+    }
+  }, [isOpen]);
+
+  const Component = getComponent('media-library').Component;
```
Comment (HichamELBSI) [inline docs/3.0.0-beta.x/guides/registering-a-field-in-admin.md:73]: You can precise the component name. Something like `MediaLibraryComponent`.

File: docs/3.0.0-beta.x/guides/registering-a-field-in-admin.md [hunk 2]
```diff
@@ -65,12 +143,28 @@ const Wysiwyg = ({
...
 }) => {
+  const [isOpen, setIsOpen] = useState(false);
   let spacer = !isEmpty(inputDescription) ? <div style={{ height: '.4rem' }} /> : <div />;
 
   if (!noErrorsDescription && !isEmpty(errors)) {
     spacer = <div />;
   }
 
+  const handleChange = data => {
+    console.log(data);
```
Comment (HichamELBSI) [inline docs/3.0.0-beta.x/guides/registering-a-field-in-admin.md:154]: console.log

File: docs/3.0.0-beta.x/guides/registering-a-field-in-admin.md [hunk 3]
```diff
@@ -65,12 +143,28 @@ const Wysiwyg = ({
...
+
+    if (data.mime.includes('image')) {
+      const imgTag = `<p><img src="${data.url}" caption="${data.caption}" alt="${data.alternativeText}"></img></p>`;
+      const newValue = value ? `${value}${imgTag}` : imgTag;
+
+      onChange({ target: { name, value: newValue } });
+    }
+
+    // Handle videos and other type of files
```
Comment (HichamELBSI) [inline docs/3.0.0-beta.x/guides/registering-a-field-in-admin.md:163]: Is this comment useful?

File: docs/3.0.0-beta.x/plugin-development/frontend-components-api.md [hunk 4]
```diff
@@ -0,0 +1,160 @@
...
+
+const Initializer = ({ updatePlugin }) => {
+  const {
+    strapi: { componentApi },
+  } = useStrapi();
+  const ref = useRef();
+  ref.current = updatePlugin;
+
+  useEffect(() => {
+    // Register the new field
```
Comment (HichamELBSI) [inline docs/3.0.0-beta.x/plugin-development/frontend-components-api.md:96]: Register the new component.

File: packages/strapi-admin/admin/src/utils/ComponentApi.js [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+
+  registerComponent = component => {
+    const { name, Component } = component;
+
+    invariant(Component, 'A Component must be provided');
+    invariant(name, 'A name must be provided');
+    invariant(this.components[name] === undefined, 'A similar field already exists');
+
+    this.components[name] = { Component };
+  };
```
Comment (HichamELBSI) [inline packages/strapi-admin/admin/src/utils/ComponentApi.js:31]: Is there a way to register multiple components?

File: packages/strapi-plugin-content-manager/admin/src/components/Wysiwyg/MediaLib.js [hunk 6]
```diff
@@ -0,0 +1,69 @@
...
+      setIsDisplayed(true);
+    }
+  }, [isOpen]);
+
+  const Component = getComponent('media-library').Component;
+
+  const handleInputChange = data => {
+    if (data) {
+      const { name, alternativeText, url } = data;
+      const alt = alternativeText === '' ? name : alternativeText;
```
Comment (HichamELBSI) [inline packages/strapi-plugin-content-manager/admin/src/components/Wysiwyg/MediaLib.js:25]: As `''` is false in JS, you just need to do `const alt = alternativeText || name;`

File: packages/strapi-plugin-content-manager/admin/src/components/WysiwygBottomControls/index.js [hunk 7]
```diff
@@ -7,37 +9,12 @@
...
-          }}
-        >
-          <FormattedMessage
-            id="components.WysiwygBottomControls.uploadFiles"
-            values={{ browse }}
-          />
-          <input type="file" onChange={onChange} />
-        </label>
-      </div>
+      <div />
```
Comment (HichamELBSI) [inline packages/strapi-plugin-content-manager/admin/src/components/WysiwygBottomControls/index.js:17]: Is this empty `div` needed?

File: packages/strapi-plugin-upload/admin/src/components/BrowseAssets/index.js [hunk 8]
```diff
@@ -67,6 +68,23 @@ const BrowseAssets = () => {
...
+    ? null
+    : id => (
+        <CardControl
+          small
+          title="edit"
+          color="#9EA7B8"
+          type="pencil"
+          onClick={() => handleGoToEditFile(id)}
+        />
+      );
```
Comment (HichamELBSI) [inline packages/strapi-plugin-upload/admin/src/components/BrowseAssets/index.js:83]: This is a bit strange. Can't we do something like 
```js
const renderCardControl = id => {
    if (noNavigation) {
        return null
    }
    
    return (
        <CardControl
          small
          title="edit"
          color="#9EA7B8"
          type="pencil"
          onClick={() => handleGoToEditFile(id)}
        />
      );
}
```

File: packages/strapi-plugin-upload/admin/src/components/ListModal/index.js [hunk 9]
```diff
@@ -33,12 +33,10 @@ const ListModal = () => {
     </BaselineAlignmentWrapper>
   );
 
+  const renderRightContent = noNavigation ? null : renderUploadModalButton;
```
Comment (HichamELBSI) [inline packages/strapi-plugin-upload/admin/src/components/ListModal/index.js:36]: Same here, can't we put the noNavigation condition in the renderUploadModalButton function?

File: docs/3.0.0-beta.x/guides/registering-a-field-in-admin.md (hunk 1 above)
Comment (soupette) [reply docs/3.0.0-beta.x/guides/registering-a-field-in-admin.md:73]: Media library is the name of the component

File: docs/3.0.0-beta.x/guides/registering-a-field-in-admin.md (hunk 2 above)
Comment (soupette) [reply docs/3.0.0-beta.x/guides/registering-a-field-in-admin.md:154]: I thought it was useful for the user to add this but I'll remove it

File: docs/3.0.0-beta.x/guides/registering-a-field-in-admin.md (hunk 3 above)
Comment (soupette) [reply docs/3.0.0-beta.x/guides/registering-a-field-in-admin.md:163]: Sure it is there to say that in the example I have only handle the image type so they will need to add the logic for other types

File: packages/strapi-admin/admin/src/utils/ComponentApi.js (hunk 5 above)
Comment (soupette) [reply packages/strapi-admin/admin/src/utils/ComponentApi.js:31]: Like in the field api noe

File: packages/strapi-plugin-content-manager/admin/src/components/WysiwygBottomControls/index.js (hunk 7 above)
Comment (soupette) [reply packages/strapi-plugin-content-manager/admin/src/components/WysiwygBottomControls/index.js:17]: I did not want to change the styling code so yes

File: packages/strapi-plugin-upload/admin/src/components/BrowseAssets/index.js (hunk 8 above)
Comment (soupette) [reply packages/strapi-plugin-upload/admin/src/components/BrowseAssets/index.js:83]: Well if you prefer but it is equivalent to not passing a fender méthode so personally I'd rather have no method called if it's not needed

**Pull request merged**